### PR TITLE
Remove quantization related config from dequantized model

### LINKED ISSUE #34847: model.dequantize() does not remove quantization_config from model.config which effects in error when loading the model
### System Info

- `transformers` version: 4.46.0
- `bitsandbytes` version: 0.44.1
- Platform: Linux-5.10.0-32-cloud-amd64-x86_64-with-glibc2.31
- Python version: 3.10.14
- Huggingface_hub version: 0.25.1
- Safetensors version: 0.4.5
- Accelerate version: 1.0.0
- Accelerate config: 	not found
- PyTorch version (GPU?): 2.4.1+cu121 (True)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using distributed or parallel set-up in script?: <fill in>
- Using GPU in script?: yes 
- GPU type: NVIDIA A100-SXM4-40GB

### Who can help?

quantization: @SunMarc @MekkCyber

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

```python
from transformers import AutoModel, AutoTokenizer
from transformers import BitsAndBytesConfig

bnb_config = {
    "bnb_4bit_compute_dtype": "bfloat16",
    "bnb_4bit_quant_type": "nf4",
    "bnb_4bit_use_double_quant": True,
    "load_in_4bit": True
}

model = AutoModel.from_pretrained("mistralai/Mistral-7B-v0.1", quantization_config=bnb_config)
model = model.dequantize()
model.save_pretrained("mistral_test_dequantized")
model = AutoModel.from_pretrained("mistral_test_dequantized")
```
Stacktrace:
```
Traceback (most recent call last):
  File "./test_model.py", line 40, in <module>
    main()
  File "./test_model.py", line 37, in main
    model = AutoModel.from_pretrained("mistral_test_dequantized")
  File "./venv/lib/python3.10/site-packages/transformers/models/auto/auto_factory.py", line 564, in from_pretrained
    return model_class.from_pretrained(
  File "./venv/lib/python3.10/site-packages/transformers/modeling_utils.py", line 4226, in from_pretrained
    ) = cls._load_pretrained_model(
  File "./venv/lib/python3.10/site-packages/transformers/modeling_utils.py", line 4729, in _load_pretrained_model
    new_error_msgs, offload_index, state_dict_index = _load_state_dict_into_meta_model(
  File "./venv/lib/python3.10/site-packages/transformers/modeling_utils.py", line 994, in _load_state_dict_into_meta_model
    hf_quantizer.create_quantized_param(model, param, param_name, param_device, state_dict, unexpected_keys)
  File "./venv/lib/python3.10/site-packages/transformers/quantizers/quantizer_bnb_4bit.py", line 207, in create_quantized_param
    raise ValueError(
ValueError: Supplied state dict for layers.0.mlp.down_proj.weight does not contain `bitsandbytes__*` and possibly other `quantized_stats` components.
```

### Expected behavior

Dequantized model can be saved using `save_pretrained` and properly loaded using `from_pretrained`

### PR DESCRIPTION
**Pull request by konradkalita**
# What does this PR do?

Fixes #34847

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

 @SunMarc

### CODE REVIEW

File: src/transformers/quantizers/base.py [hunk 1]
```diff
@@ -215,7 +215,10 @@ def dequantize(self, model):
 
         # Delete quantizer and quantization config
         del model.hf_quantizer
-
+        del model.config.quantization_config
+        del model.config._pre_quantization_dtype
+        model.is_quantized = False
```
Comment (konradkalita) [inline src/transformers/quantizers/base.py:220]: Quantizer seems better place for it than PreTrainedModel.

Not sure if that's 100% exhaustive, but it was enough to solve my issue.

Comment (SunMarc) [reply src/transformers/quantizers/base.py:220]: Nice, sounds good to me !

**Review by SunMarc — APPROVED** [review]
Nice ! Could you add a test in the bnb tests to check if it works as expected ?

**Review by ArthurZucker — APPROVED** [review]
makes sense, thanks for updating

**Pull request merged**